bean-utils-library/src/main/java/com/hotels/beans/transformer/TransformerImpl.java

```diff
@@ -315,21 +315,22 @@ private String getDestFieldName(final Parameter constructorParameter, final Stri
         Function<Object, Object> transformerFunction = getTransformerFunction(field, fieldBreadcrumb);
         boolean isTransformerFunctionDefined = nonNull(transformerFunction);
         Object fieldValue = getSourceFieldValue(sourceObj, sourceFieldName, isTransformerFunctionDefined);
+        Object outputValue = fieldValue;
         if (nonNull(fieldValue)) {
             // is not a primitive type or an optional && there are no transformer function
             // defined it recursively evaluate the value
             boolean notPrimitiveAndNotSpecialType = !primitiveType && !classUtils.isSpecialType(fieldType);
             if (!isTransformerFunctionDefined
                     && (notPrimitiveAndNotSpecialType || Optional.class.isAssignableFrom(fieldValue.getClass()))) {
-                fieldValue = getFieldValue(targetClass, field, fieldValue, fieldBreadcrumb);
+                outputValue = getFieldValue(targetClass, field, fieldValue, fieldBreadcrumb);
             }
         } else if (primitiveType) {
```

Comment:
I don't think you need all those changes to achieve the goal, maybe it would be enough to modify this condition adding: 
```
} else if (primitiveType && !isTransformerFunctionDefined) {
```
what do you think?